lib/base-compiler.ts

```diff
@@ -1739,7 +1742,7 @@ export class BaseCompiler implements ICompiler {
         const makeGccDump =
             backendOptions.produceGccDump && backendOptions.produceGccDump.opened && this.compiler.supportsGccDump;
 
-        const downloads = await this.setupBuildEnvironment(key, dirPath, filters.binary);
+        const downloads = await this.setupBuildEnvironment(key, dirPath, filters.binary || filters.binaryobject);
```

Comment:
same question as for the liblink[] arguments

Reply:
I don't really know what's hidden behing `setupBuildEnv`. Is it really only for linking? Reading it now let me think it really sets up the env for using any library, so this can impact codegen (e.g. something from a header file). Is it not the case?

Reply:
Ahh right, for c++ it's not needed because we store the header files on disk and that's handled by https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/base-compiler.ts#L817 

But for Rust, yes, it would be needed. Maybe it needs an override in rust.ts?

Reply:
I'm Ok with leaving this as-is, and we speed it up at a later point (-> it doesn't do any harm, it just has to wait for possible downloads)

Reply:
Ok, thanks, I'll leave it then! As for Rust, I still need to check the whole feature works